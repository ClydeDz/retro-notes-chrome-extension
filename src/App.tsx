import "./styles/App.css";
import RnUserInput from "./components/RnUserInput";
import { Elements } from "./constants/Elements";
import RnFooter from "./components/RnFooter";
import RnHeader from "./components/RnHeader";

function App() {
  return (
    <div className="app">
      <RnHeader />
      <main>
        {Elements.map(element => {
            return (
              <div className="col" key={element.componentKey}>
                <RnUserInput text={element.text} componentKey={element.componentKey} />
              </div>
            );
        })}
      </main>
      <RnFooter />
    </div>
  );
}

export default App;
