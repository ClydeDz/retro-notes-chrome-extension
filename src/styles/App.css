.app {
  padding: 9px 18px;
  width: 800px;
  margin: 10px auto;
}

.col {
  height: 200px;
  width: 257px;
  float: left;  
  margin-bottom: 9px;
}

textarea {
  resize: none;
  height: 150px;
  width: 95%;
  background-color: #ffffff12; 
  color: white;
  border: 1px solid #282c34;
  outline: 0;
  border-radius: 0;
  padding: 2px 4px;
}

textarea:focus {
  border: 1px solid #61dafb65;
}
