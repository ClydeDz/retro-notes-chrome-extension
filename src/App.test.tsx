import { render, screen } from "@testing-library/react";
import App from "./App";
import { chrome } from "jest-chrome";

// @ts-expect-error
global.chrome = chrome;

describe("<App />", () => {
  test("renders the main elements of the app", () => {
    render(<App />);
    const wellElement: HTMLElement = screen.getByText(/What went well/i);
    const didntGoWellElement: HTMLElement = screen.getByText(/What didn't go well/i);
    const notesElement: HTMLElement = screen.getByText(/Notes/i);

    expect(wellElement).toBeInTheDocument();
    expect(didntGoWellElement).toBeInTheDocument();
    expect(notesElement).toBeInTheDocument();
  });
})

