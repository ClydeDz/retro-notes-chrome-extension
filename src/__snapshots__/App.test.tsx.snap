// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> snapshot testing the default view 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <header
          class="App-header"
        >
          <div
            class="col"
          >
            <div
              aria-label="User input"
            >
              <h2
                aria-label="Heading"
              >
                What went well?
              </h2>
              <textarea
                aria-label="Text area"
              />
            </div>
          </div>
          <div
            class="col"
          >
            <div
              aria-label="User input"
            >
              <h2
                aria-label="Heading"
              >
                What didn't go well?
              </h2>
              <textarea
                aria-label="Text area"
              />
            </div>
          </div>
          <div
            class="col"
          >
            <div
              aria-label="User input"
            >
              <h2
                aria-label="Heading"
              >
                Notes
              </h2>
              <textarea
                aria-label="Text area"
              />
            </div>
          </div>
        </header>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <header
        class="App-header"
      >
        <div
          class="col"
        >
          <div
            aria-label="User input"
          >
            <h2
              aria-label="Heading"
            >
              What went well?
            </h2>
            <textarea
              aria-label="Text area"
            />
          </div>
        </div>
        <div
          class="col"
        >
          <div
            aria-label="User input"
          >
            <h2
              aria-label="Heading"
            >
              What didn't go well?
            </h2>
            <textarea
              aria-label="Text area"
            />
          </div>
        </div>
        <div
          class="col"
        >
          <div
            aria-label="User input"
          >
            <h2
              aria-label="Heading"
            >
              Notes
            </h2>
            <textarea
              aria-label="Text area"
            />
          </div>
        </div>
      </header>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
